crates/cdk/src/mint/issue/mod.rs

```diff
@@ -547,156 +553,420 @@ impl Mint {
-        // verify the signature is provided for the mint request
-        if let Some(pubkey) = mint_quote.pubkey {
-            mint_request.verify_signature(pubkey)?;
+            total_available_amount = total_available_amount
+                .checked_add(quote_amount)
+                .ok_or(Error::AmountOverflow)?;
+            mintable_per_quote.push(quote_amount);
         }
 
         let Verification {
             amount: outputs_amount,
-            unit,
-        } = match self.verify_outputs(&mut tx, &mint_request.outputs).await {
+            unit: output_unit,
+        } = match self.verify_outputs(&mut tx, &batch_request.outputs).await {
             Ok(verification) => verification,
             Err(err) => {
                 tracing::debug!("Could not verify mint outputs");
-
                 return Err(err);
             }
         };
 
-        if mint_quote.payment_method == PaymentMethod::Bolt11 {
-            // For bolt11 we enforce that mint amount == quote amount
-            if outputs_amount != mint_amount {
-                return Err(Error::TransactionUnbalanced(
-                    mint_amount.into(),
-                    mint_request.total_amount()?.into(),
-                    0,
-                ));
-            }
-        } else {
-            // For other payments we just make sure outputs is not more then mint amount
-            if outputs_amount > mint_amount {
+        if payment_method == PaymentMethod::Bolt11 {
+            if outputs_amount != total_available_amount {
                 return Err(Error::TransactionUnbalanced(
-                    mint_amount.into(),
-                    mint_request.total_amount()?.into(),
+                    total_available_amount.into(),
+                    batch_request.total_amount()?.into(),
                     0,
                 ));
             }
+        } else if outputs_amount > total_available_amount {
+            return Err(Error::TransactionUnbalanced(
+                total_available_amount.into(),
+                batch_request.total_amount()?.into(),
+                0,
+            ));
         }
 
-        let unit = unit.ok_or(Error::UnsupportedUnit)?;
-        ensure_cdk!(unit == mint_quote.unit, Error::UnsupportedUnit);
+        // Allocate how much each quote actually mints: bolt11 must issue full amount, bolt12 caps at remaining.
+        let minted_amounts_per_quote = match payment_method {
+            PaymentMethod::Bolt11 => mintable_per_quote.clone(),
+            PaymentMethod::Bolt12 => {
+                let mut remaining = outputs_amount;
+                let mut per_quote = Vec::with_capacity(mintable_per_quote.len());
+
+                for available in &mintable_per_quote {
+                    let minted = std::cmp::min(*available, remaining);
+                    per_quote.push(minted);
+                    remaining = remaining.checked_sub(minted).ok_or(Error::AmountOverflow)?;
+                }
+
+                per_quote
```

Comment:
I think this could be problematic if a large payment came in while the wallet is making the request it did not account for and that is the first quote in the list then all the outputs would be allocated minting for that quote and leaving none for the rest of the quotes. Worse the wallet has no way of knowing this even happened without further checks of mint quotes. 

Should the wallet state in the request how much it expects for each quote.